src/include/esp-idf/bindings.h

```diff
@@ -157,4 +157,8 @@
 #include "host/util/util.h"
 #endif
 
+#include "mdns.h"
```

Comment:
Is mdns a separate ESP-IDF  component? If so, you probably need to protect it with `#ifdef` conditionals whether that component is enabled (not all ESP-IDF components are enabled by default for `cmake`-first builds and then register the component here: https://github.com/esp-rs/esp-idf-sys/blob/master/build/common.rs#L31

Also, I strongly suggest to make a separate PR for these additional includes, as I can merge them right away (provided that you address the components' issue), and they do not have to wait for the critical section thing, which is more complex.